Adding Run 330098 (2019 MWGR3) in data relvals

boudoul opened the pull request:
#### PR description:
Greetings
This PR aims to introduce a new run 3 data workflow based on fresh run3 data taken during MWGR3 .
The goal is to test the run3 reconstruction with real data and anticipate any issue without waiting for the usual Tier0 replay prior the global runs - Eventually we can update with more fresh runs once we get some.


#### PR validation:
I ran the following workflow runTheMatrix.py -l 138.1 --command "-n 100"  

#### if this PR is a backport please specify the original PR:
It may need to be backported in 10_6 (depending on the MWGR 4 plan) but let's make a review in master first

boudoul committed:
Adding Run 328788 (2019 MWGR1) in data relvals

boudoul commented:
I would need : 
- comments from PdmV  (@pgunnell,@prebello,@zhenhu) : if they are happy with the way I'm introducing this WF ( WF number, naming etc..) 
- comments from Alca ( @christopheralanwest,@franzoni,@tlampen,@pohsun,@tocheng )  : I  used a candidate GT kindly prepared by @christopheralanwest a couple of weeks ago, and defined it as run3_data in autocond. Would it be possible to get a 'real GT' ( so that I can update this PR with it and not leave a candidate GT in this workflow )  - Thanks.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27341/10598

- This PR adds an extra 52KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/AlCa/python/autoCond.py  modified in PR(s): #27334
  - File  Configuration/PyReleaseValidation/python/relval_standard.py  modified in PR(s): #26476
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #26476, #27336

cmsbuild commented:
A new Pull Request was created by  @boudoul (boudoul) for master.

It involves the following packages:

Configuration/AlCa
Configuration/PyReleaseValidation


@cmsbuild, @prebello, @zhenhu, @christopheralanwest, @tocheng, @pgunnell, @franzoni, @kpedro88, @tlampen, @pohsun can you please review it and eventually sign? Thanks.
@makortel, @Martin-Grunewald, @mmusich, @tocheng this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mmusich commented on Configuration/AlCa/python/autoCond.py:
```diff
@@ -44,6 +44,8 @@
     # GlobalTag for Run2 HLT for HI (not 2018 HI): it points to the online GT
     'run2_hlt_hi'       :   '101X_dataRun2_HLTHI_frozen_v9',
     # GlobalTag for MC production with perfectly aligned and calibrated detector for Phase1 2017 (and 0,0,~0-centred beamspot)
+    'run3_data'         :   '106X_dataRun3_Express_Candidate_2019_05_24_21_36_45',
```
@boudoul in the past 'run2_data' key was reserved for offline / re-reco Global Tags. I think to preserve the nomenclature something like `run3_data_promptlike` would be more appropriate. By the way do we have a Run3 prompt queue?

christopheralanwest replied:
I agree that `run3_data` should be reserved for offline global tags. 

@boudoul Do we plan to have prompt reco for the MWGRs or only express? We do have both express and prompt queues for Run 3: "106X_dataRun3_Express_Queue" and "106X_dataRun3_Prompt_Queue". But if we are not planning to have prompt reco in the near future, we should have an express GT here: there's no point in validating a GT that we don't plan to use. I would be inclined to use the key name `run3_data_promptlike` in either case.

fabiocos commented:
The overlap with #27336 is not going to produce conflicts

christopheralanwest commented on Configuration/AlCa/python/autoCond.py:
```diff
@@ -44,6 +44,8 @@
     'run2_hlt_hi'       :   '101X_dataRun2_HLTHI_frozen_v9',
     # GlobalTag for MC production with perfectly aligned and calibrated detector for Phase1 2017 (and 0,0,~0-centred beamspot)
+    'run3_data'         :   '106X_dataRun3_Express_Candidate_2019_05_24_21_36_45',
+    # GlobalTag for Run3 data relvals
```
The comments in L46 and L48 should be switched so that the comment is above the corresponding key-value pair. Please try to maintain the same whitespace in the unchanged lines to avoid irrelevant merge conflicts.

boudoul replied:
Hi @christopheralanwest , thanks for your comments (sorry for the delay, CMS week..) We should consider having prompt reconstruction in MWGR, do you have a name of a GT I should use  ?  (I will also use un3_data_promptlike as you recommended)   - And for the line switching , thanks for spotting , it was not intentionnal to change , I will fix in a next commit. Thanks again .

boudoul commented:
Dear  Alca ( @christopheralanwest,@franzoni,@tlampen,@pohsun,@tocheng )- Could you please let me know the name of the GT I should use, in order for me to edit this PR with this new GT and also take care of the comments at the same time. MWGR is approaching .... 
Thank you .

tlampen commented:
Dear Gaëlle, my understanding is that  it should be 106X_upgrade2021_realistic_v8 . I hope @christopheralanwest can confirm to be sure.

boudoul commented:
Hi @tlampen , thanks. This sounds more like a MC GT and not a data GT, no ? Anyway I let @christopheralanwest comment when he can in these busy days .

christopheralanwest commented:
We currently don't have any better data GT for MWGR#1 than the candidate `106X_dataRun3_Express_Candidate_2019_05_24_21_36_45` that you have proposed. When MWGR#1 was taken, proper Run-3 HCAL conditions for data were not yet available. The candidate GT referenced here includes HCAL test tags with IOV from [1, inf) so that a test of the new HCAL conditions could be performed. But because of synchronization rules needed to ensure reproducibility, we cannot append new IOVs to a tag with an express synchronization with an IOV later than (last run taken at P5+1).

cmsbuild commented:
The code-checks are being triggered in jenkins.

boudoul commented:
ok thanks @christopheralanwest - I believe I fixed in my last comment autocond according to your previous comments

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27341/10722

- This PR adds an extra 56KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_standard.py  modified in PR(s): #26476, #27415
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #26476, #27415, #27392

cmsbuild commented:
Pull request #27341 was updated. @cmsbuild, @prebello, @zhenhu, @christopheralanwest, @tocheng, @pgunnell, @franzoni, @kpedro88, @tlampen, @pohsun can you please check and sign again.

christopheralanwest commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 43d05999cbbaf5961311f80eba70b39f0535f571
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a1ff85/1294/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a1ff85/1294/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3290686
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3290351
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

prebello commented:
please test workflow 138.1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 43d05999cbbaf5961311f80eba70b39f0535f571

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a1ff85/1298/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
138.1 step2
<pre>runTheMatrix-results/138.1_RunCosmics2019+RunCosmics2019+RECOCOSDRUN3+ALCACOSDRUN3+HARVESTDCRUN3/step2_RunCosmics2019+RunCosmics2019+RECOCOSDRUN3+ALCACOSDRUN3+HARVESTDCRUN3.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

christopheralanwest commented:
The change of `auto:run3_data` to `auto:run3_data_promptlike` needs to be propagated to the other files in the commit.

The maintenance of this new relval will present some awkward issues as the proper conditions for this run cannot be provided when following the usual rules for handling prompt queues. There are several options:

- Generate new HCAL tags that introduce the HCAL conditions for the HB upgrade, setting the last IOV to some time prior to run 328788. This violates the usual requirement that the history in prompt queues be preserved. This is only an option because prompt reco has never been run on 2019 data. If we do not do this, it will not be possible to reconstruct MWGR#2 data except with this ad hoc candidate. On the other hand, it is likely have no one will ever look at MWGR#2 data after this month.
- Change the run number in the future so that it points to some run from MWGR#3 instead.
- Do not merge this PR but only use it for offline tests and reconsider it once MWGR#3 data is available.

Which is the preferred option? I'm also adding @abdoulline to this discussion to get an opinion from HCAL.

abdoulline commented:
Thank you for adding me to the discussion, Chris.
To me arranging RelVal for MWGR#3 would be the preferred option, if there is no particular importance(rush) of including MWGR#1 specifically...

fabiocos commented:
@boudoul as we are having several concurrent updates proposed for PyReleaseValidation , what is the plan for this one? From the thread I understand that there are issues that would suggest to keep this on hold until MWGR2

boudoul commented:
Hi @fabiocos ,  I failed to make understand the goal of this PR to the participants, which was actually to run in an IB the only data we have so far (we cannot invent new ones before the run) in order to test in IBs the workflow *before* a replay (Tier0 is not meant to validate a workflow, it should be tested first...) . But at this point I will anyway request a replay likely today with this run  and we will see . We will decide what do do later with this PR. 

So we can put it on hold (and anyhow I will need to fix the auto:run3_data_promptlike that I forgot to propagate) - Thanks

fabiocos commented:
hold

cmsbuild commented:
Pull request has been put on hold by  @fabiocos
They need to issue an `unhold` command to remove the `hold` state or L1 can `unhold` it for all

christopheralanwest commented:
I guess this PR could now be revisited using new data from the latest MWGR and the prompt GT 106X_dataRun3_Prompt_v2?

boudoul commented:
Hi @christopheralanwest , definitely - I will update with a fresh run and the GT (on my list for tomorrow..)

cmsbuild commented:
The code-checks are being triggered in jenkins.

boudoul commented:
Hello @christopheralanwest - After a rebase, I have updated the GT , fixed the run3_data_promptlike , use a fresh run 330098 and updated title & doc - At your convenance , could you please look at it ? Thanks

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27341/11069

- This PR adds an extra 52KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/AlCa/python/autoCond.py  modified in PR(s): #27600, #27604, #27517
  - File  Configuration/PyReleaseValidation/python/relval_standard.py  modified in PR(s): #26476
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #26476, #27433

cmsbuild commented:
Pull request #27341 was updated. @cmsbuild, @prebello, @zhenhu, @christopheralanwest, @tocheng, @pgunnell, @franzoni, @kpedro88, @tlampen, @pohsun can you please check and sign again.

christopheralanwest commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: b1bc188467aa86d8f0ebd42c38cf34ac9ba57397
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a1ff85/1652/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a1ff85/1652/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 2628546
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2628227
 - DQMHistoTests: Total skipped: 317
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

christopheralanwest commented:
please test workflow 138.1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: b1bc188467aa86d8f0ebd42c38cf34ac9ba57397
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a1ff85/1659/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a1ff85/1659/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-a1ff85/138.1_RunCosmics2019+RunCosmics2019+RECOCOSDRUN3+ALCACOSDRUN3+HARVESTDCRUN3

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 2628546
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2628227
 - DQMHistoTests: Total skipped: 317
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

christopheralanwest commented:
+1

boudoul commented:
Thanks @christopheralanwest !
@fabiocos , can we unhold now ? and if the remaining signatures (@prebello, @zhenhu, @pgunnell, @kpedro88) could also take a look , that's would be great. Thanks.

kpedro88 commented:
+upgrade

kpedro88 commented:
unhold

boudoul commented:
Hi PdmV  (@prebello, @zhenhu, @pgunnell ), we need your comments here and eventually your signature

prebello commented:
+1

all looks ok. 138.1 tested and approved. sorry for the delay @boudoul. This github announcement skipped my daily check.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

boudoul commented:
Thanks @prebello !

kpedro88 commented:
+1
